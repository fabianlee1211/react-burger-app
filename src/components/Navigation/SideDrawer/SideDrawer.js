import React from 'react';
import Logo from '../../Logo/Logo';
import NavigationItems from '../NavigationItems/NavigationItems';
import Backdrop from '../../UI/Backdrop/Backdrop';
import classes from './SideDrawer.css';

const SideDrawer = (props) => {
  const { open, closed } = props
  let attachedClasses = [classes.SideDrawer, classes.Close]
  if(open) {
    attachedClasses = [classes.SideDrawer, classes.Open];
  }
  return (
    <React.Fragment>
      <Backdrop show={open} clicked={closed} />
      <div className={attachedClasses.join(' ')}>
        <div className={classes.Logo}><Logo /></div>
        <nav onClick={closed}>
          <NavigationItems />
        </nav>
      </div>
    </React.Fragment>
  );
};

export default SideDrawer;