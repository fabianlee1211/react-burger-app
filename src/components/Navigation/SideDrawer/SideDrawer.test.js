import React from 'react';
import SideDrawer from './SideDrawer';
import { shallow } from 'enzyme';

describe('<SideDrawer />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<SideDrawer />);
  });

  it('renders SideDrawer correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders a Backdrop', () => {
    expect(wrapper.find('Backdrop')).toHaveLength(1);
  });

  it('renders an opened side menu with backdrop', () => {
    wrapper.setProps({ open: true });
    const backdrop = wrapper.find('Backdrop');
    expect(wrapper.find('div.SideDrawer.Open')).toHaveLength(1);
    expect(backdrop.props().show).toEqual(true);
  });

  it('renders a closed side menu without backdrop', () => {
    wrapper.setProps({ open: false });
    const backdrop = wrapper.find('Backdrop');
    expect(wrapper.find('div.SideDrawer.Close')).toHaveLength(1);
    expect(backdrop.props().show).toEqual(false);
  });

  it('toggles SideDrawer', () => {
    const clickFn = jest.fn();
    wrapper.setProps({ closed: clickFn });
    const backdrop = wrapper.find('Backdrop');
    expect(backdrop.props().clicked).toBe(clickFn);
    wrapper.find('nav').simulate('click');
    expect(clickFn).toHaveBeenCalled();
  });
});
