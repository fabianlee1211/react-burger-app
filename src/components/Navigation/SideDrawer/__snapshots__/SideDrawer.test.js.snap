// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideDrawer /> renders SideDrawer correctly 1`] = `
<Fragment>
  <Backdrop />
  <div
    className="SideDrawer Close"
  >
    <div
      className="Logo"
    >
      <Logo />
    </div>
    <nav>
      <NavigationItems />
    </nav>
  </div>
</Fragment>
`;
